app/src/main/java/com/example/minerva/data/local/ArticleDao.kt

```diff
@@ -0,0 +1,20 @@
+package com.example.minerva.data.local
+
+import androidx.room.*
+import com.example.minerva.data.model.Article
+import kotlinx.coroutines.flow.Flow
+
+@Dao
+interface ArticleDao {
+    @Insert(onConflict = OnConflictStrategy.IGNORE)
+    suspend fun insertListOfArticles(articles: List<Article>)
+
+    @Update
+    suspend fun insertFavouriteArticle(favouriteArticle: Article)
+
+    @Query("SELECT * FROM articles WHERE email = :userEmail")
+    fun getAllFavouriteArticlesByUserEmail(userEmail: String): Flow<List<Article>>
+
+    @Delete
```

Comment:
ummmm, I think this also should be update? I mean what you are trying to do here is change the boolean to false not delete the whole record?

Reply:
```
@Update
  suspend fun insertFavouriteArticle(favouriteArticle: Article)
```
  
  I think this will be sufficient for handling both cases, What do you think ?

Reply:
Yup